From 2ceaa05a5a3aa3e8b9c67877eb968b2813fb4d18 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Tue, 7 Oct 2014 14:30:34 -0400
Subject: gdm: fix sensitivity of auth prompt when cancelling early and user
 list is disabled

If the user list is disabled and the user clicks cancel quickly enough
after typing their username, they can get in a state where the
auth prompt gets stuck in the insensitive state.

This is because the login dialog code makes the prompt insensitive
while while pam is processing the provided username, but the prompt
only makes itself sensitive again when it is hidden.

This commit makes it sensitive right before asking for a username again.

https://bugzilla.gnome.org/show_bug.cgi?id=740141

diff --git a/js/gdm/loginDialog.js b/js/gdm/loginDialog.js
index ceaccaa..6a8ae0a 100644
--- a/js/gdm/loginDialog.js
+++ b/js/gdm/loginDialog.js
@@ -643,6 +643,8 @@ const LoginDialog = new Lang.Class({
                                                         realmManager.release();
                                                     }));
         this._updateCancelButton();
+
+        this._authPrompt.updateSensitivity(true);
         this._showPrompt();
     },
 
-- 
cgit v0.10.1

